File: solutions/haskell/secret-handshake/1/src/SecretHandshake.hs
module SecretHandshake (handshake) where
import Data.Bits ((.&.))

handshake :: Int -> [String]
handshake num = 
  let 
    bit x = (num .&. (2^x)) /= 0
    res0 = (if bit 0 then ["wink"] else [])
      ++ (if bit 1 then ["double blink"] else [])
      ++ (if bit 2 then ["close your eyes"] else [])
      ++ (if bit 3 then ["jump"] else [])
  in if bit 4 then reverse res0 else res0  

File: solutions/python/secret-handshake/1/secret_handshake.py
WORDS = ["jump", "close your eyes", "double blink", "wink"]

def commands(binary_str):
    res_list = [WORDS[i] for i in range(4) if binary_str[i + 1] == "1"]
    if binary_str[0] == "0":
        res_list.reverse()
    return res_list